src/libs/error.mock.ts

```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash/fp'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+export type ErrorExports = typeof import('src/libs/error') & { __esModule: true }
+
+export const errorWatcher = jest.fn()
+export const mockErrorModule = (): ErrorExports => {
```

Comment:
this does a good job of following the recommended Shared Mock pattern in the unit testing guide

https://docs.google.com/document/d/1bZ1uc6uSQ-CzydcVRDhcKpAqFIGh13yZNzu_lrSffqM/edit#heading=h.rrj2ecwl8ihk

But for mocks within a utility file like this, we should offer mocks of individual things that a test may want to mock, instead of the entire module as is being mocked here.  This here should export mockWithErrorReportingInModal and move the module assembly near line 24 below to the test file.  It's a bit more "assembly required", but offers the desired composition for cases where a test file may want/need to mock more then one of the exports from a utility file like this.

Reply:
also, please add a jsdoc block for this export with usage example once updated with pattern tweak above.

Reply:
Hmm.. my thought was that most consumers would want to mock all `withErrorXXX` constructs. I can split them out though so its more composable.